lib/datadog/core/remote/component.rb

```diff
@@ -20,23 +21,36 @@ def initialize(settings, agent_settings)
           transport_options = {}
           transport_options[:agent_settings] = agent_settings if agent_settings
 
+          negotiation = Negotiation.new(settings, agent_settings)
           transport_v7 = Datadog::Core::Transport::HTTP.v7(**transport_options.dup)
 
           capabilities = Client::Capabilities.new(settings)
 
           @barrier = Barrier.new(BARRIER_TIMEOUT)
 
           @client = Client.new(transport_v7, capabilities)
+          healthy = false
+          Datadog.logger.debug { "new remote configuration client: #{@client.id}" }
+
           @worker = Worker.new(interval: settings.remote.poll_interval_seconds) do
+            unless healthy || negotiation.endpoint?('/v0.7/config')
+              @barrier.lift
+
+              next
+            end
```

Comment:
It looks like we want to lift the barrier only when the client is unhealthy and the agent does not respond to the config endpoint: 

```ruby
irb(main):059:1* unless false || true
irb(main):060:1*   puts "hello"
irb(main):061:0> end
=> nil
irb(main):062:1* unless true || true
irb(main):063:1*   puts "hello"
irb(main):064:0> end
=> nil
irb(main):065:1* unless true || false
irb(main):066:1*   puts "hello"
irb(main):067:0> end
=> nil
irb(main):068:1* unless false || false
irb(main):069:1*   puts "hello"
irb(main):070:0> end
hello
```

Are we doing that to unblock callers of `Datadog::Core::Remote.active_remote.barrier(:once)` in case the agent does not support remote configuration?

Reply:
> when the client is unhealthy and the agent does not respond to the config endpoint

Correct: `¬(A ∧ B) ⇔ ¬A ∨ ¬B`

Reply:
> Are we doing that to unblock callers of Datadog::Core::Remote.active_remote.barrier(:once) in case the agent does not support remote configuration?

Yep:

- It'll try to negotiate again and again until it's "healthy"
- a client is not considered "healthy" until the first sync happens
- If a client fails in such a way that we must recreate one, then it's back to the initial state and thus considered not healthy, and will try to negotiate again (because the failure may be that the agent was stopped or dropped RC)